src/AutoMapper/QueryableExtensions/ProjectionExpression.cs

```diff
@@ -104,7 +105,10 @@ internal class MemberVisitor : ExpressionVisitor
         {
             protected override Expression VisitLambda<T>(Expression<T> node)
             {
-                var memberExpression = node.Body as MemberExpression;
+                MemberExpression memberExpression = 
+                    node.Body.NodeType == ExpressionType.Convert 
+                        ? (node.Body as UnaryExpression)?.Operand  as MemberExpression
+                        :  node.Body as MemberExpression;
```

Comment:
No. If to remove this fix, you'll get NullReferenceException on passing lambda of accessing a member of value type, after the line `var memberExpression = node.Body as MemberExpression;`. as operator returns null due to top level body is other than `MemberExpression`, namely, Convert operation. You can make sure that it is true by reversing this hunk and running/debugging the tests from this PR (kind of TDD approach professing). This is a bug of crash level that forces using ProjectTo signature with **string[]** notation of **membersToExpand** that makes compile-time control on field names more difficult and less reliable.

This is because value types need **boxing** to be assigned to **object** target. In ExpressionTree, it is expressed in wrapping it to Expression.Convert which is actually UnaryExpression with one operand that actually is MemberExpression with return type of value type. If we not convert, it is not assignable to `Expression<Func<Source, object>>`.

`Expression<Func<Dto, int?  >>` is not assignable to `Expression<Func<Dto, object>>`. You cannot pass both . In case of code in tests, the compiler infers them all as E`xpression<Func<Dto, object>>` and adds Expression.Convert where necessary. The code below demonstrates it.

```
Expression<Func<Dto, int?  >> nullableMemberExpression1 = _ => _.Desc; Console.WriteLine(nullableMemberExpression1); // _ => _.Desc
Expression<Func<Dto, object>> nullableMemberExpression2 = _ => _.Desc; Console.WriteLine(nullableMemberExpression2); // _ => Convert(_.Desc)
Expression<Func<Dto, object>> referenceMemberExpression = _ => _.Name; Console.WriteLine(referenceMemberExpression); // _ => _.Name

// Expression<Func<Dto, object>> projectExpression = nullableMemberExpression1;
// error CS0029: Cannot implicitly convert type 'System.Linq.Expressions.Expression<System.Func<AutoMapper.UnitTests.Projection.MembersToExpand_ExpressionArray.Dto, int?>>' to 'System.Linq.Expressions.Expression<System.Func<AutoMapper.UnitTests.Projection.MembersToExpand_ExpressionArray.Dto, object>>'

// _iq.ProjectTo<Dto>(Configuration, referenceMemberExpression, nullableMemberExpression1);
// error CS1503: Argument 4: cannot convert from 'System.Linq.Expressions.Expression<System.Func<AutoMapper.UnitTests.Projection.MembersToExpand_ExpressionArray.Dto, int?>>' to 'System.Linq.Expressions.Expression<System.Func<AutoMapper.UnitTests.Projection.MembersToExpand_ExpressionArray.Dto, object>>'

_iq.ProjectTo<Dto>(Configuration, nullableMemberExpression1).First().Desc.ShouldEqual(_iqf.Desc);
_iq.ProjectTo<Dto>(Configuration, referenceMemberExpression, nullableMemberExpression2);

```